src/object/cli_obj.c

```diff
@@ -617,16 +617,157 @@ obj_shard_find_replica(struct dc_object *obj, unsigned int target,
 }
 
 static int
-obj_grp_leader_get(struct dc_object *obj, int idx, unsigned int map_ver, uint8_t *bit_map)
+obj_ec_leader_select(struct dc_object *obj, int grp_idx, uint8_t *bit_map)
```

Comment:
hmm, ec object will go obj_list_get_shard->obj_ec_list_get_shard(),  then for to_leader case, it will only choose the last parity.

if (obj_auxi->to_leader) {
                        /* proper way to choose leader ? XXX */
                        shard = parities[0];
                        D_GOTO(out, shard);
                }

This is a bit messy indeed, but I will cleanup further in the following patch.

Reply:
ah, ok, thanks.